tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -448,6 +452,60 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
     }
 
+    /* Test scenario: client connection is wiped + HRR + server sends cookie data as part of HRR
+     *
+     * Tests that calls to `s2n_connection_wipe` are safe and do not cause the connection to end up in
+     * a bad state. s2n_connection_wipe is called when a customer wants to reuse a connection; which is
+     * more performant compared to creating a new connection. */
+    {
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+        /* ensure call to s2n_connection_wipe are safe */
+        EXPECT_SUCCESS(s2n_connection_wipe(client_conn));
+        EXPECT_SUCCESS(s2n_connection_wipe(server_conn));
+
+        DEFER_CLEANUP(struct s2n_config *server_config = s2n_config_new(),
+                s2n_config_ptr_free);
+        DEFER_CLEANUP(struct s2n_config *client_config = s2n_config_new(),
+                s2n_config_ptr_free);
+        DEFER_CLEANUP(struct s2n_cert_chain_and_key *tls13_chain_and_key = s2n_cert_chain_and_key_new(),
+                s2n_cert_chain_and_key_ptr_free);
+
+        char tls13_cert_chain[S2N_MAX_TEST_PEM_SIZE] = {0};
+        char tls13_private_key[S2N_MAX_TEST_PEM_SIZE] = {0};
+
+         /* Create nonblocking pipes */
+        struct s2n_test_io_pair io_pair;
+        EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+
+        EXPECT_SUCCESS(s2n_read_test_pem(S2N_ECDSA_P384_PKCS1_CERT_CHAIN, tls13_cert_chain, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_read_test_pem(S2N_ECDSA_P384_PKCS1_KEY, tls13_private_key, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_cert_chain_and_key_load_pem(tls13_chain_and_key, tls13_cert_chain, tls13_private_key));
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(client_config, tls13_chain_and_key));
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, tls13_chain_and_key));
```

Comment:
Nit: Since these test files were written, we simplified how to load test certs and keys because it was so verbose / repetitive. See https://github.com/aws/s2n-tls/blob/main/tests/testlib/s2n_testlib.h#L169-L170